File: backend/node-express/src/strategies/DateOnlyStrategy.ts
// NOTE: THIS CLASS DOES NOT NEED MODIFIED

import { RouteStrategy, MatchWithCity, OptimisedRoute } from './RouteStrategy';
import { buildRoute } from '../utils/buildRoute';

/**
 * DateOnlyStrategy — A naive route optimisation that simply sorts by date.
 *
 * This strategy produces valid routes but doesn't consider geographic distance.
 * It's provided as a working example so you can test your endpoints before
 * implementing NearestNeighbourStrategy.
 *
 * Try running this strategy and then your NearestNeighbourStrategy on the
 * same set of matches — compare the total distances!
 */
export class DateOnlyStrategy implements RouteStrategy {
  optimise(matches: MatchWithCity[]): OptimisedRoute {
    // Simply sort matches by kickoff time — no distance optimisation
    const sorted = [...matches].sort(
      (a, b) => new Date(a.kickoff).getTime() - new Date(b.kickoff).getTime()
    );

    return buildRoute(sorted, 'date-only');
  }
}

File: backend/python-flask/app/strategies/date_only_strategy.py
# NOTE: THIS CLASS DOES NOT NEED MODIFIED

from app.strategies.route_strategy import RouteStrategy, build_route


class DateOnlyStrategy(RouteStrategy):
    """
    DateOnlyStrategy — A naive route optimisation that simply sorts by date.

    This strategy produces valid routes but doesn't consider geographic distance.
    It's provided as a working example so you can test your endpoints before
    implementing NearestNeighbourStrategy.

    Try running this strategy and then your NearestNeighbourStrategy on the
    same set of matches — compare the total distances!
    """

    def optimise(self, matches: list) -> dict:
        # Simply sort matches by kickoff time — no distance optimisation
        sorted_matches = sorted(matches, key=lambda m: m['kickoff'])
        return build_route(sorted_matches, 'date-only')
